flags.go

```diff
@@ -5,29 +5,68 @@
 package main
 
 import (
+	"crypto/tls"
+	"errors"
 	"flag"
+	"fmt"
 	"os"
 	"strings"
 )
 
-func parseFlags() {
-	parseFlagSetWithArgs(flag.CommandLine, os.Args)
+var supportedTLSVersions map[string]uint16 = map[string]uint16{
+	"1.0": tls.VersionTLS10,
+	"1.1": tls.VersionTLS11,
+	"1.2": tls.VersionTLS12,
+	"1.3": tls.VersionTLS13,
 }
 
-func parseFlagSetWithArgs(flagSet *flag.FlagSet, args []string) {
-	flagsFromEnv(flagSet)
+type tlsMinVersionValue struct {
+	version     *string
+	versionCode *uint16
+}
+
+func (t tlsMinVersionValue) String() string {
+	if t.version != nil {
+		return *t.version
+	}
+	return ""
+}
```

Comment:
If we store only the code in the value, we could have also a map here to convert it back to string.
```suggestion
func (t tlsMinVersionValue) String() string {
   switch tlsMinVersionValue {
   case tls.VersionTLS10:
     return "1.0"
   case tls.VersionTLS11:
     return "1.1"
   case tls.VersionTLS12:
     return "1.2"
   case tls.VersionTLS13:
     return "1.3"
   default:
     return ""   
   }
}
```